kernel/src/table_changes/scan.rs

```diff
@@ -0,0 +1,204 @@
+use std::sync::Arc;
+
+use itertools::Itertools;
+use tracing::debug;
+
+use crate::scan::ColumnType;
+use crate::schema::SchemaRef;
+use crate::{DeltaResult, ExpressionRef};
+
+use super::{TableChanges, CDF_FIELDS};
+
+/// The result of building a [`TableChanges`] scan over a table. This can be used to get a change
+/// data feed from the table
+#[derive(Debug)]
+#[allow(unused)]
```

Comment:
don't need since it's pub?
```suggestion
```

Reply:
do we need it?

Reply:
Ah sry I missed this. All fields are unused. Should I do `allow(unused)` for each field instead?

Reply:
Ohhh i see - the fields - nope this is fine :) just didn't realize that before!